### ISSUE TITLE
`qiskit.test` is going to be removed in `qiskit 1.0`

### ISSUE THREAD

**Issue by 1ucian0** [Qiskit/documentation#145]
Qiskit exposes test utilities as part of its deliverable package. This is not consider good practice and [its removal is a long overdue issue](https://github.com/Qiskit/qiskit/issues/6862). Qiskit teams's intention is [to clean that up for `qiskit 1.0`](https://github.com/Qiskit/qiskit/pull/10998) (planned to be released on Jan 31st, 2024).  The module will be available in the `0.*` versions (EoL, mid 2024) and [raise a deprecation warning](https://github.com/Qiskit/qiskit/pull/11001).

The documentation uses `qiskit.test` in, at least, the following file:
https://github.com/Qiskit/documentation/blob/main/docs/api/qiskit-ibm-runtime/runtime_service.md?plain=1#L33

Consider stop importing `qiskit.test` (the IBM specific `FakeProviders` are moving to [IBM packages](https://github.com/Qiskit/qiskit-ibm-provider/pull/740)).

**Comment by frankharkins** [comment]
@1ucian0 is this true for `qiskit.test.reference_circuits`? If so, do you know where that's moving?

**Comment by 1ucian0** [comment]
yes, they are getting moved to `test/utils/reference_circuits.py`, which is not part of the package. Do you need them for something?

**Comment by frankharkins** [comment]
We're using it but we can easily switch it out, thanks!
